File: src/app_state.rs
use std::collections::HashMap;
use std::fs::File;
use std::io::BufReader;

use crate::travel_serializer::Bus;

#[derive(Debug)]
pub struct AppState {
    pub bus: Bus,
    pub bus_stops: HashMap<i64, String>,
}

impl AppState {
    pub fn init() -> Self {
        let file = File::open("travel.json").expect("file doesn't find");
        let reader = BufReader::new(file);
        let bus: Bus = serde_json::from_reader(reader).expect("failed to parse json to bus");

        let mut bus_stops: HashMap<i64, String> = HashMap::new();
        for bus in bus.bus_line.iter() {
            for stop in bus.travel.iter() {
                if bus_stops.get(&stop.id).is_none() {
                    bus_stops.insert(stop.id, stop.name.clone());
                }
            }
        }

        Self { bus, bus_stops }
    }
}

#[test]
fn test_app_stat() {
    let app_state = AppState::init();
    println!("{app_state:?}");
}

File: src/core/app_state.py
from typing import List

from core.utils import BusLine, load_bus_data


class AppState:
    def __init__(self):
        self.buslines: List[BusLine] = load_bus_data("data/travel.json")
        self.travels = {
            str(travel.id): travel.name
            for busline in self.buslines
            for travel in busline.travel
        }
        self.travel_sets = {
            bus_line.id: {t.id for t in bus_line.travel} for bus_line in self.buslines
        }

        self.caches = {}
